scientific_library/tvb/simulator/monitors.py

```diff
@@ -291,8 +293,8 @@ def config_for_sim(self, simulator):
         # setup given spatial mask or default to region mapping
         if self.spatial_mask is None:
             self.is_default_special_mask = True
-            if not (simulator.surface is None):
-                self.spatial_mask = simulator.surface.region_mapping
+            if simulator.surface is not None:
+                self.spatial_mask, _, _ = self.backend.full_region_map(simulator.surface, simulator.connectivity)
```

Comment:
Could it be that the `backend. full_region_map` implementation is missing from the current committed state ?

Reply:
it's in the branch [here](https://github.com/BalanceKey/tvb-root/blob/reg-map-fix/scientific_library/tvb/simulator/backend/ref.py#L112)